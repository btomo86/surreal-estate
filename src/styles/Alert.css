.alert-success {
  font-size: 20px;
  color: rgb(1, 97, 1);
}

.alert-failure {
  font-size: 20px;
  color: red;
}
