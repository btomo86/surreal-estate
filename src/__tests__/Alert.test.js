import React from "react";
import { render } from "@testing-library/react";
import Alert from "../components/Alert";

test("displays an error message", () => {
  const { getByText } = render(<Alert message="Error!" />);
  expect(getByText(/Error/).textContent).toBe("Error!");
});

test("displays a success message", () => {
  const { getByText } = render(<Alert message="Success!!!!" success />);
  expect(getByText(/Success/).textContent).toBe("Success!!!!");
});

test("does not render an error or a success message if message props is empty", () => {
  const { getByText, asFragment } = render(<Alert message="Error!" />);
  const alert = asFragment();
  expect(alert).toMatchSnapshot();
  expect(asFragment()).toMatchSnapshot();
  expect(getByText(/Error/).textContent).toBe("Error!");
});
