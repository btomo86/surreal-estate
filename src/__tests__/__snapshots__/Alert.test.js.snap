// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render an error or a success message if message props is empty 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`does not render an error or a success message if message props is empty 2`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;
